### ISSUE TITLE
[Request] Advanced Prompting Option for LobeChat

### ISSUE THREAD

**Issue by xenstar** [lobehub/lobe-chat#2816]
### 🥰 Feature Description

The proposed feature is an advanced prompting option in LoBeChat that allows users to customize their prompts with specific requirements quicker without needing multiple agents. This feature aims to enhance the user experience by providing a more tailored and efficient way to interact with the chatbot.

### 🧐 Proposed Solution

Add an "Advanced Prompting" icon in the chatbox that, when hovered over, displays a dropdown menu with four options:

1. Output Format
2. Tone
3. Writing Style
4. Language

![image](https://github.com/lobehub/lobe-chat/assets/55502651/a0468ce2-9320-4645-a938-1b9cc40e8a3a)

Each option has a list of sub-options (detailed below) that users can select from.

When a user selects one or more options, the prompt is rewritten to include the chosen specifications. For example:

Benefits of magnesium glycinate. (this is actual promt), below it will add

Please respond in [Tone], [Writing Format], [Language]. Answer in [Output Format]

If you are confused about it, let me know, I can make a video to explain it, how it will work.

### 📝 Additional Information

Detailed Lists for Each Option:

**Output Format:**

Concise
Step-by-step
Extreme Detail
ELI5
Essay
Report
Summary
Table
FAQ
Listicle
Interview
Review
News
Opinion
Tutorial
Case Study
Profile
Blog
Poem
Script
Whitepaper
Ebook
Press Release
Infographic
Webinar
Podcast Script
Email Campaign
Social Media Post
Proposal
Brochure
Newsletter
Presentation
Product Description
Research Paper
Speech
Memo
Policy Document
User guide
Technical documentation
Q&A
First Principles Thinking
Feynman Technique

**Tone:**

Authoritative
Clinical
Cold
Confident
Cynical
Emotional
Empathetic
Formal
Friendly
Humorous
Informal
Ironic
Optimistic
Pessimistic
Playful
Sarcastic
Serious
Sympathetic
Tentative
Warm

**Writing Style:**

Academic
Analytical
Argumentative
Conversational
Creative
Critical
Descriptive
Epigrammatic
Epistolary
Expository
Informative
Instructive
Journalistic
Metaphorical
Narrative
Persuasive
Poetic
Satirical
Technical

**Language:**

English
Spanish
French
German
Chinese
Japanese
Indonesian
Bengali
Korean
Russian
Arabic
Portuguese

**Comment by xenstar** [comment]
Since lobehub have plugins, it can be developed as a plugin. It still gives better output, and for new users it's quite helpful..

**Comment by arvinxx** [comment]
I think this type of requirement should now be addressed through skills, right?

---
> This comment was translated by Claude.

<details>
<summary>Original Content</summary>
这类需求我觉得在现在来看应该是都通过 skill 来解决？
</details>

**Comment by xenstar** [comment]
> I think this type of requirement should now be addressed through skills, right?

I feel like today Arvin is treating me as VVIP user. Haha (after almost 1.5 year) :P 


yes, no need now, as I have setup a lot of different agents for different task execution.
